controllers/LoginController.php

```diff
@@ -3,37 +3,51 @@
 
 class LoginController
 {
-    public static function index()
+    private $_repo;
+
+    public function __construct(IRepository $repository)
+    {
+        $this->_repo = $repository;
+    }
+
+    public function index()
     {
         echo LoginIndexView::render();
-        return;
     }
 
-    public static function set()
+    public function set()
     {
-        print_r($_REQUEST);
-        
-        $userRepository = new UserRepository();
+        if (!isset($_SESSION['uid']) || $_SESSION['uid'] == '') {
+            $exist = $this->_repo->exist($_REQUEST);
 
-        $user = $userRepository->findUserToLogin($_REQUEST['username'], $_REQUEST['password']);
+            if ($exist) {
+                $user = $this->_repo->getByUsername($_REQUEST['username']);
 
-        print_r($user->__toString());
+                $_SESSION['uid'] = hash("md5", $user);
 
-        $_SESSION['uid'] = $user->getId();
+                echo "Successfully logged in for: " . $user . PHP_EOL;
 
-        die("Tu jest ustawianie sesji");
+                // TODO: Redirect to other page
+                // echo invoiceSaleIndexView::render();
+            }
+            else {
+                echo "Wrong credentials" . PHP_EOL;
+                // echo LoginIndexView::render();
+            }
+        }
+        else {
+            echo "You are already logged in." . PHP_EOL;
+            // echo LoginIndexView::render();
+        }
 
-        // TODO: Przejście do zalogowanej strony
     }
 
     public static function logout()
     {
-        $suc1 = session_unset();
-        $suc2 = session_destroy();
+        session_unset();
+        session_destroy();
 
-        if ($suc1 && $suc2) {
-            die("Wylogowano!");
-        }
+        echo "Successfully logged out." . PHP_EOL;
```

Comment:
Tak samo trzeba też widoki, przekierowania do stron po podaniu poprawnych danych, wylogowaniu itp zamiast tych 'echo'.

Reply:
wydaje mi sie ze szkoda na to czasu, narazie samo przekierowanie wystarczy